Framework/Indexing/src/IndexInfo.cpp

```diff
@@ -309,6 +319,21 @@ IndexInfo::globalSpectrumIndicesFromDetectorIndices(
       auto bytes = static_cast<int>(sizeof(int64_t) * spectrumIndices.size());
       communicator().send(rank, tag, buffer, bytes);
     }
+    if (std::any_of(detectorMap.begin(), detectorMap.end(),
+                    [](const std::pair<char, std::vector<char>> &p) {
+                      return p.first == 1;
+                    })) {
+      throw std::runtime_error("Some of the requested detectors do not have a "
+                               "corresponding spectrum");
+    }
+    if (std::any_of(detectorMap.begin(), detectorMap.end(),
+                    [](const std::pair<char, std::vector<char>> &p) {
+                      return std::any_of(p.second.begin(), p.second.end(),
+                                         [](char c) { return c > 2; });
```

Comment:
Again I find this a bit confusing, why is it checking `> 2`? I think it would be easier to understand if you put https://github.com/mantidproject/mantid/pull/22790/files?utf8=%E2%9C%93&diff=split#diff-8f01dcea0e18444bf24f13c378566f22R306 as an `else` to https://github.com/mantidproject/mantid/pull/22790/files?utf8=%E2%9C%93&diff=split#diff-8f01dcea0e18444bf24f13c378566f22R299, and check `> 1` here?

Reply:
👍